crates/subspace-networking/src/node_runner.rs

```diff
@@ -293,9 +294,18 @@ where
                 endpoint,
                 ..
             } => {
+                let shared = match self.shared_weak.upgrade() {
+                    Some(shared) => shared,
+                    None => {
+                        return;
+                    }
+                };
```

Comment:
Just a note. We can do this right now:
```rust
let Some(shared) = self.shared_weak.upgrade() else { return };
```

Reply:
That is exactly what I wrote initially. Unfortunately `rustfmt` seems to be unable to format `let .. else`, which is why I'm not using it yet anywhere. I'll be checking with new Rust releases as we upgrade and will start using it once formatting works.